### PR TITLE
Move domain access fields on node forms

### PR DESCRIPTION
**Pull request by bjaxelsen**
I've made this based on #239. It allows us to configure Domain Access to move the Domain Access fields to an advanced tab on the node forms. This is good for editors who may not understand the meaning of the fields - they can concentrate on the main content fields.

I don't think this interferes in a bad way with the field group module. The advanced groups on the node edit forms is already available and by making this a configurable option it can easily be turned off in case you want to move the fields to some other field group etc.

### CODE REVIEW

File: domain_access/config/install/domain_access.settings.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+node_advanced_tab: 'TRUE'
```
Comment (zerolab) [inline domain_access/config/install/domain_access.settings.yml]: This needs to be simply `true`.

The tests fail with:

```
Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for
    domain_access.settings with the following errors:
    domain_access.settings:node_advanced_tab variable type is string but
    applied schema class is Drupal\Core\TypedData\Plugin\DataType\BooleanData
    in Drupal\Core\Config\Testing\ConfigSchemaChecker-&gt;onConfigSave()
```

**Comment by zerolab** [comment]
Checked this on my local test build. Does what it says on the tin. When the setting is enabled, the fields will move to the advanced groups even if they are added to a field group in the form display settings. The only downside is that the field group then appears as empty, with just the title.

I think it would be helpful to add a note on the domain_access settings page about that to avoid any confusion.

**Comment by bjaxelsen** [comment]
Yes, I think it makes sense to only move the fields if they are not put in a field group already. I've changed this and modified the help text on the configuration form.

**Comment by zerolab** [comment]
I am not entirely sure about this. The core fields that are in the advanced tab do not react to them being in a field group. So maybe this should do the same.

File: domain_access/domain_access.routing.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+domain_config.admin:
```
Comment (agentrickard) [inline domain_access/domain_access.routing.yml:1]: This route is for domain_access, not domain_config.

File: domain/domain.links.task.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
-entity.domain.edit_form:
```
Comment (agentrickard) [inline domain/domain.links.task.yml:1]: I don't understand why this task is being removed here.

**Comment by agentrickard** [comment]
OK, I've moved this into the general settings branch that I'm working on, Thanks!

File: domain/domain.links.task.yml (hunk 3 above)
Comment (bjaxelsen) [reply domain/domain.links.task.yml:1]: sorry, think I made a mistake here

Comment (agentrickard) [reply domain/domain.links.task.yml:1]: It doesn't seem to affect anything. See the new PR in https://github.com/agentrickard/domain/pull/264

**Pull request closed without merging**